### PR TITLE
react-native: Add support for Pad and TVOS in Platform

### PR DESCRIPTION
**Pull request by alexmngn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/blob/master/Libraries/Utilities/Platform.ios.js
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -5614,7 +5614,7 @@ export interface PixelRatioStatic {
  */
 export type PlatformOSType = 'ios' | 'android' | 'windows' | 'web'
 
-interface PlatformStatic {
+interface PlatformStatic extends PlatformIOSStatic {
```
Comment (alloy) [inline types/react-native/index.d.ts]: Shouldn’t this be the other way around and require the user to cast the `Platform` module to the iOS one if `OS = 'ios'` ?

Comment (alexmngn) [reply types/react-native/index.d.ts]: I'm actually not sure how to manage this piece with TS as Platform is for both but those attribute are only available for iOS. 

Should we have something like this?

```
export var Platform: PlatformStatic | PlatformIOSStatic;
```

Comment (alloy) [reply types/react-native/index.d.ts]: I was thinking something like the following:

```ts
interface PlatformIOSStatic extends PlatformStatic { ... }
```

```ts
import * as ReactNative from "react-native"

const Platform: PlatformIOSStatic = ReactNative.Platform
```

But I have no idea which would be more usable. Maybe give both options a try and feel which works best?

**Comment by alexmngn** [comment]
@alloy Made the change. Basically anyone who needs isTV or isPad will have to override the Platform as suggested.

**Comment by alloy** [comment]
Gotcha, I think that makes sense 👍 Thanks!

**Pull request merged**